flutter_local_notifications_windows/lib/src/details/notification_details.dart

```diff
@@ -0,0 +1,162 @@
+  alarm,
+
+  /// Calls are expanded and show in a special format.
+  ///
+  /// By default, call notifications loop the standard "call" sound.
+  incomingCall,
+
+  /// Urgent notifications can break through Do Not Disturb settings.
+  urgent,
+}
+
+extension on DateTime {
+  String toIso8601StringTz() {
+    // Get offset
+    final offset = timeZoneOffset;
+    final sign = offset.isNegative ? "-" : "+";
+    final hours = offset.inHours.abs().toString().padLeft(2, "0");
+    final minutes = offset.inMinutes.abs().remainder(60).toString().padLeft(2, "0");
+    final offsetString = "$sign$hours:$minutes";
+    // Get first part of properly formatted ISO 8601 date
+    final formattedDate = toIso8601String().split(".").first;
+    return "$formattedDate$offsetString";
+  }
+}
+
+/// Contains notification details specific to Windows.
+///
+/// See: https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/adaptive-interactive-toasts
+class WindowsNotificationDetails {
+  /// Creates a Windows notification from the given options.
+  const WindowsNotificationDetails({
+    this.actions = const <WindowsAction>[],
+    this.inputs = const <WindowsInput>[],
+    this.images = const <WindowsImage>[],
+    this.groups = const <WindowsRow>[],
+    this.progressBars = const <WindowsProgressBar>[],
+    this.bindings = const {},
+    this.header,
+    this.audio,
+    this.duration,
+    this.scenario,
+    this.timestamp,
+    this.subtitle,
+  });
+
+  /// A list of at most five action buttons.
+  final List<WindowsAction> actions;
+
+  /// A list of at most five input elements.
+  final List<WindowsInput> inputs;
+
+  /// A custom audio to play during this notification.
+  final WindowsNotificationAudio? audio;
+
+  /// The duration for this notification.
+  final WindowsNotificationDuration? duration;
+
+  /// The scenario for this notification. Sets some defaults based on the value.
+  final WindowsNotificationScenario? scenario;
+
+  /// The header for this group of notifications.
+  final WindowsHeader? header;
+
+  /// Overrides the timestamp to show on the notification.
+  final DateTime? timestamp;
+
+  /// A third line to show under the notification body.
+  final String? subtitle;
+
+  /// A list of images to show.
+  final List<WindowsImage> images;
+
+  /// A list of groups to show.
+  final List<WindowsRow> groups;
+
+  /// A list of progress bars to show.
+  final List<WindowsProgressBar> progressBars;
+
+  /// Custom bindings in the notification.
+  ///
+  /// Text elements can contains "bindings", which are entered as `{bindingName}` directly into the
+  /// string values. You can then update them while or after the notification is launched by
+  /// using the binding name as the key here, and the value as any string you want
+  final Map<String, String> bindings;
+
+  /// XML attributes for the toast notification as a whole.
+  Map<String, String> get attributes => {
```

Comment:
From what I could tell, it looks like `attributes`, `toXml()` and `generateBindings()` is really only for internal use when the plugin converts the notification to the XML representation. However, all three are currently publicly exposed. I suspect these be made hidden and would reduce the chance of an inadvertent breaking change being made

Reply:
Looks like there are other classes in a similar situation and I left a related comment on `notification_to_xml.dart'. Is it possible to move the conversion/mapping logic to that file and avoid exposing the methods in that class?

Reply:
I'll try to see if there's a better way to hide these on Monday

Reply:
I put these in a private extension